Set up the application required files

dennisja opened the pull request:
created run.py for running the application
created app/models.py to hold the application models
created app/__init__.py to initialize the application
created app/views.py to hold the applicatioon views
created app/decorators.py to hold the required decorators
created config.py to keep track of the application configurations
created db_create.py to enable database creation
created db_upgrade.py to handle database upgrading
created db_migrate.py to haandle database migrations
created db_downgrade.py to handle database downgrades
created tests.py to handle application unit tests
created yummy.log for loggind some application specific information

dennisja committed:
Set up the application required files
created run.py for running the application
created app/models.py to hold the application models
created app/__init__.py to initialize the application
created app/views.py to hold the applicatioon views
created app/decorators.py to hold the required decorators
created config.py to keep track of the application configurations
created db_create.py to enable database creation
created db_upgrade.py to handle database upgrading
created db_migrate.py to haandle database migrations
created db_downgrade.py to handle database downgrades
created tests.py to handle application unit tests
created yummy.log for loggind some application specific information

dennisja commented:
@patrickcmd, @matthewacha my pull request

dennisja committed:
Added modules
Added pytest for unit tests
Added sql-alchemy flask extension

dennisja committed:
Modified requirements.txt
This had non required dependancies which where removed when it was updated

PatrickCmd commented:
I see your trying to implement databases. Please take note on Challenge 2 NO.16 and the self-assessment guideline of this challenges. For now data structures(lists, dictionaries etc) and OOP concepts are needed to implement this challenge.
Please take note.
The UI design is great

dennisja committed:
Changes made
Modified the index.html to rplace the lastname placeholder with a name example
Modified db_upgrade.py, db_downgrade.py, db_upgrade.py to log infor to the yummy.log file
Wrote tests for the validator
Wrote the validation class in app/validator.py to validate form input

daisyndungu commented on app/__init__.py:
```diff
@@ -0,0 +1,9 @@
+app.config.from_object("config")
+db = SQLAlchemy(app)
+
+from app import views, models
```
Add a new line at the end of this file

dennisja replied:
Ok, i will @daisyndungu

matthewacha commented on config.py:
```diff
@@ -0,0 +1,8 @@
+
+basedir = os.path.abspath(os.path.dirname(__file__))
+# path to the database
+SQLALCHEMY_DATABASE_URI = 'sqlite:///'+os.path.join(basedir,"app.db")
```
use "app_db"or "app" instead of  "app.db" , it automatically creates the database and adds .db

dennisja replied:
Thanks @matthewacha

matthewacha commented on run.py:
```diff
@@ -0,0 +1,4 @@
+from app import app
+
+if __name__ == "__main__":
```
remove line 3, its redundant

matthewacha commented:
check the files you created, i left comments on specific lines

dennisja committed:
Removed database related files
Removed all files related to the database
Added simulators.py to contain classes that will simulate a user, Users, recipes, and recipes categories
Wrote the Users class
Wrote tests for the Users class
Wrote tests for the user class
Wrote the User class
Wrote a decorator to disable access of some view functions when a user is logged in
Wrote tests for the validator class
Wrote a validator class in validator.py to validate form input

dennisja committed:
Fixed all errors
Fixed all errors which were in the tests
Removed the models.py file since we do not need it at this stage

dennisja committed:
Linked the home page user interface to the backend
Fixed all the errors appearing in tests
Modified the views, to enable user registration and login
The login is not yet validated, any values can login
The registration is not yet persistent, your values will be lost
Currently the recipes data is fetched from the RecipesList class
No functionality for adding a recipe neither a category
